data-science-onramp/data-cleaning/clean_test.py

```diff
@@ -0,0 +1,158 @@
+                "PIP_PACKAGES": "google-cloud-storage"
+            },
+        },
+        'master_config': {
+            'num_instances': 1,
+            'machine_type_uri': 'n1-standard-8'
+        },
+        'worker_config': {
+            'num_instances': 6,
+            'machine_type_uri': 'n1-standard-8'
+        },
+        "initialization_actions": [
+            {
+                "executable_file": ("gs://dataproc-initialization-actions/"
+                                    "python/pip-install.sh"),
+            }
+        ],
+        "software_config": {
+            "image_version": "1.5.4-debian10",
+            "optional_components": [
+                "ANACONDA"
+            ],
+        }
+    }
+}
+
+@pytest.fixture(autouse=True)
+def setup_and_teardown_cluster():
+    # Create cluster using cluster client
+    cluster_client = dataproc.ClusterControllerClient(client_options={
+        'api_endpoint': f'{REGION}-dataproc.googleapis.com:443'
+    })
+    operation = cluster_client.create_cluster(PROJECT, REGION, CLUSTER_DATA)
+
+    # Wait for cluster to provision
+    operation.result()
+
+    yield
+
+    # Delete cluster
+    operation = cluster_client.delete_cluster(PROJECT, REGION, CLUSTER_NAME)
+    operation.result()
+
+@pytest.fixture(autouse=True)
+def setup_and_teardown_bucket():
+    # Create GCS Bucket
+    storage_client = storage.Client()
+    bucket = storage_client.create_bucket(BUCKET_NAME)
+
+    # Upload file
+    blob = bucket.blob(DESTINATION_BLOB_NAME)
+    blob.upload_from_filename("clean.py")
+
+    yield
+
+    # Delete GCS bucket
+    bucket = storage_client.get_bucket(BUCKET_NAME)
+    bucket.delete(force=True)
+
+def is_in_table(value, out):
+    return re.search(f"\\| *{value} *\\|", out)
+
+def get_blob_from_path(path):
+    bucket_name = re.search("dataproc.+?/", path).group(0)[0:-1]
+    bucket = storage.Client().get_bucket(bucket_name)
+    output_location = re.search("google-cloud-dataproc.+", path).group(0)
+    return bucket.blob(output_location)
+
+def test_clean():
+    '''Tests clean.py by submitting it to a dataproc cluster'''
+
+    # Submit job to dataproc cluster
+    job_client = dataproc.JobControllerClient(client_options={
+        'api_endpoint': f'{REGION}-dataproc.googleapis.com:443'
+    })
+    response = job_client.submit_job_as_operation(project_id=PROJECT, region=REGION, job=JOB_DETAILS)
+
+    # Wait for job to complete
+    result = response.result()
+
+    # Get job output
+    output_location = result.driver_output_resource_uri + ".000000000"
+    blob = get_blob_from_path(output_location)
+    out = blob.download_as_string().decode("utf-8")
+
+    # tripDuration
+    assert not is_in_table("(\\d+(?:\\.\\d+)?) s", out)
+    assert not is_in_table("(\\d+(?:\\.\\d+)?) min", out)
+    assert not is_in_table("(\\d+(?:\\.\\d+)?) h", out)
+
+    # station latitude & longitude
+    assert not is_in_table("[0-9]+" + u"\u00B0" + "[0-9]+\'[0-9]+\"", out)
```

Comment:
Maybe we should add an assert that there are floating point values for longitude and latitude?

Reply:
good idea